### PR TITLE
Tiered Dungeons rebalance by Eric

### PR DESCRIPTION
**Pull request by eric-anderson**
### All Submissions:

- [ YES ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### Tested Changes:

1. [ YES ] Have you lint your code (lua lint) locally prior to submission?

### Changes to Core Features:

- [ YES ] Have you added an explanation of what your changes do and why you'd like us to include them?

See commit messages. For overall summary of the set of changes see below (where I copied it) or 

https://github.com/eric-anderson/ComfyFactorio/blob/75864b385c38d2cb191d22a3bed207858d7d1179/maps/dungeons/tiered_dungeon.lua#L872

Which is the information for dungeons in the new changelog tab.

- [ YES ] Have you successfully ran tests with your changes locally?

Server is currently running if you want to try it out.

### Comments

	{ ver = '1.1', date = '2022-03-13', desc = [[
* All research is now found at random.
  * Red science floors 0-1
  * Green on floors 1-5
  * Gray on floors 5-10
  * Blue on floors 8-13
  * Blue/gray on floors 10-14
  * Purple on floors 12-19
  * Yellow on floors 14-21
  * White on floors 20-25
  * Atomic Bomb/Spidertron on floors 22-25
* Add melee mode toggle to top bar. Keeps weapons in main inventory if possible.
* Ore from rocks nerfed. Used to hit max value on floor 2, now scales up from
  floors 0-19 along with ore from rooms. After floor 20 ore from rooms scales up faster.
* Treasure rooms
  * Rescaled to have similar total resources regardless of size
  * Unlimited number of rooms but lower frequency
  * Loot is limited to available loot 3 floors lower, but slightly more total value than before.
* Autostash and corpse clearing from Mountain Fortress enabled
* Harder rooms will occur somewhat farther out on the early floors.
* Spawners and worm counts bounded in early rooms.
]]},

### CODE REVIEW

**Comment by Gerkiz** [comment]
Made some GUI changes. Moved the files over to .../utils/gui/. Hence the conflict.

File: maps/dungeons/tiered_dungeon.lua
Comment (hanakocz) [inline maps/dungeons/tiered_dungeon.lua:21]: What is the reasoning to include Collapse module there, just to disable it and never use it again?

File: utils/commands/misc.lua
Comment (Gerkiz) [inline utils/commands/misc.lua]: It stays trusted only.

File: utils/commands/misc.lua
Comment (Gerkiz) [inline utils/commands/misc.lua]: This can be removed.

File: modules/collapse.lua
Comment (Gerkiz) [inline modules/collapse.lua]: Collapse "enabled/disabled" is already checked on the function, so this can be removd.

File: modules/collapse.lua
Comment (Gerkiz) [inline modules/collapse.lua]: Same as line 275.

File: modules/changelog.lua
Comment (Gerkiz) [inline modules/changelog.lua]: This module does not make much sense, we have map info to list these things.

File: modules/autostash.lua
Comment (Gerkiz) [inline modules/autostash.lua]: This does not belong here. Don't leave any comments.

File: comfy_panel/config.lua
Comment (Gerkiz) [inline comfy_panel/config.lua]: No idea what you are doing here - spaghett is not enabled unless a player admin enables it.

This can be removed.

File: modules/collapse.lua
Comment (Gerkiz) [inline modules/collapse.lua]: Same goes with this one.

**Review by Gerkiz — CHANGES_REQUESTED** [review]
Do fix these changes before this PR can be approved.

I don't understand your reasoning with adding new modules when your main objective was to "fix" Dungeons?

Next time, new stuff goes in separate PR.

File: modules/autostash.lua
Comment (hanakocz) [reply modules/autostash.lua]: I think that this autostash code is not needed in general, allowing the transport chests as valid containers is probably not the best idea in general (players should have good control what goes into them, and there are loaders available in most of scenarios), but even then, including blue chest entity type in allowed chests would be clever way (even if conditional for one map) instead of checking for coordinates of each scanned chest.

File: maps/dungeons/tiered_dungeon.lua
Comment (eric-anderson) [reply maps/dungeons/tiered_dungeon.lua:21]: modules.rpg.main requires modules.explosives
modules.explosives requires modules.collapse

After about a week of running the server, collapse started trying to collapse tiered dungeons and by the time I noticed I had a 12G log file. So in 
https://github.com/eric-anderson/ComfyFactorio/commit/73b5d03276e8202f62689e7e39d5c934365e94c4

I added the ability to disable collapse entirely and in this commit I disabled collapse.

I didn't feel confident enough in understanding the tree of dependencies in the modules to make a change elsewhere.

File: modules/collapse.lua
Comment (eric-anderson) [reply modules/collapse.lua]: Can you tell me where it's checked? As mentioned above my experience was that after a week of the server up it started log spamming about not being able to collapse.

File: modules/changelog.lua
Comment (eric-anderson) [reply modules/changelog.lua]: I wanted to separate between the changes that were made to a particular map and the information about a map. I can certainly modify it so that I can put a changelog at the end of the map. It seemed nice to let people playing know what had changed since the last version.

File: modules/autostash.lua
Comment (eric-anderson) [reply modules/autostash.lua]: This approach seemed to match with the way people tended to play "dump loot into the transport chests so it goes up to a sorter"

I can certainly remove it. In previous runs of the map I found myself manually dumping things into the transport chests since there was no easy way to auto load them. I guess the next closest would be have a chest and a loader and always pick up the loader, empty stuff and then put the loader back.

The reason for specific coordinates rather than all blue chests is to avoid the problem of dumping into blue chests used for logistics factories.

File: comfy_panel/config.lua
Comment (eric-anderson) [reply comfy_panel/config.lua]: If this isn't here, then the spaghett module keeps re-enabling the logistic systems technology when you pass through
https://github.com/eric-anderson/ComfyFactorio/blob/75864b385c38d2cb191d22a3bed207858d7d1179/comfy_panel/config.lua#L83

**Comment by eric-anderson** [comment]
> I don't understand your reasoning with adding new modules when your main objective was to "fix" Dungeons?

The starting "fix" was the research thing, the remainder came as I was playtesting and kept running into issues. The pull request came when I had a version that I hadn't found a problem with in several days.

> Next time, new stuff goes in separate PR.

Do you want me to reslice the commits?  I asked in the comfy dungeons channel and didn't get a response so sent the pull request in the first of the three options, it sounds like you would have preferred the third. "Let me know if you'd like to send the pull request with the full original chain of commits (each one was a checkpoint when I thought I'd achieved a goal, and in general each commit mostly worked), or if you'd like me to restructure it somehow (by file; single commit; separate commits for groups of changes, e.g. research, melee mode, changelog, balancing, bugfixes) ."

File: utils/commands/misc.lua
Comment (eric-anderson) [reply utils/commands/misc.lua]: resolved in 26f34ea8d5d108a6cbd20848b24bc02525c5ff98

File: utils/commands/misc.lua
Comment (eric-anderson) [reply utils/commands/misc.lua]: resolved in 26f34ea8d5d108a6cbd20848b24bc02525c5ff98

File: modules/autostash.lua
Comment (hanakocz) [reply modules/autostash.lua]: Blue Chest is special entity, it only shares graphics with the logistic blue chest. It is creative-only chest (same as red chest, both are used for transporting stuff for that unique reason). So if we do "if dungeons then blue chest is whitelisted", then it is one operation and it will automatically find the chest.

The whitelist still is just first step, the "if chest is used (e.q. has inserters), that still can discard the chest again from being eligible.

Comment (eric-anderson) [reply modules/autostash.lua]: oh, I didn't realize those chests were a different entity. Yea, should be easy to switch over to filter for those.  I still want to ignore having inserters/loaders on them since you will but dump to those is really nice.

Comment (eric-anderson) [reply modules/autostash.lua]: Fixed to use red-chest and blue-chest in 7cc268fab6a2342d52d60b6366cdf0138d93bca9

File: maps/dungeons/tiered_dungeon.lua
Comment (eric-anderson) [reply maps/dungeons/tiered_dungeon.lua:21]: I took another look at it. It looks like if I call Collapse.start_now(false) it will also disable collapse.  It's slightly less safe than the other approach since if dungeons on_init runs before collapse on_init then it won't work right, but based on some manual testing it seems to do the right thing.  Hopefully it won't break after a week.

File: modules/collapse.lua
Comment (eric-anderson) [reply modules/collapse.lua]: Removed, I found another way to do this by calling Collapse.start_now(false)

File: modules/collapse.lua
Comment (eric-anderson) [reply modules/collapse.lua]: Removed.

File: modules/collapse.lua
Comment (eric-anderson) [reply modules/collapse.lua]: Also reverted.

File: modules/changelog.lua
Comment (Gerkiz) [reply modules/changelog.lua]: Valid reason, ty!

File: comfy_panel/config.lua
Comment (Gerkiz) [reply comfy_panel/config.lua]: I'd still want to redo the trigger function instead of adding this.
But I understand now why you are adding it here.

Looks good for now.

**Review by Gerkiz — APPROVED** [review]

**Comment by Gerkiz** [comment]
Looks good. Approved the changes!

@hanakocz has the final word 👍

**Review by hanakocz — APPROVED** [review]
Think it is good to go now too, hopefully. :) Good job and thanks

**Pull request merged**
